beluga-ssh-service/src/lib.rs

```diff
@@ -1,58 +1,63 @@
 use core::net::SocketAddrV4;
+use std::net::Ipv4Addr;
 
 use beluga_tunnel::{Error, Result, Service};
 use tokio::io::{AsyncReadExt, AsyncWriteExt};
-use tokio::net::TcpStream;
+use tokio::net::{TcpListener, TcpStream};
 use tokio::sync::mpsc::{Receiver, Sender};
 
-#[derive(Debug, Default)]
-pub struct SshService;
+#[derive(Clone, Copy, Debug)]
+pub struct SshService {
+    port: u16,
+}
+
+impl SshService {
+    pub const fn source() -> Self {
+        SshService { port: 8022 }
```

Comment:
Where does this magic value come from?

Also:

```suggestion
        Self { port: 8022 }
```
(this is usually a clippy lint, but it was moved to `clippy::nursery` because of false positives)

Reply:
I figured it would be a good idea to provide a default port - the user can still use its own port by defining the struct by him or herself though

Reply:
Makes sense, but can we explicitly document that?

Something like:
```
/// Creates an SSH service using the default port.
///
/// The port is currently arbitrarily chosen by this crate to `8022`.
/// You can choose an arbitrary port by creating your own instance:
///
/// ```rust
/// SshService {
///   port: 2222
/// }
/// ```
```

Personally, I really don't like when crates make this kind of arbitrary decision for the user, but as long as it's not the only way I can live with it.